### PR TITLE
Avoiding unnecessary render

### PR DESCRIPTION
**Pull request by stevemao**
Please see my inline annotation.

### CODE REVIEW

**Comment by cameronbourke** [comment]
Hey @stevemao, `shouldComponentUpdate` is not called on the initial render, but more importantly currently you are only checking for a change in `value` to decide whether to update the component or note. What if the user changed another prop, say `minWidth` for example, the component would not reflect the new `minWidth`.

**Comment by atran** [comment]
+1 This fork helped me with a `Maximum call stack` error when setting the value from a Redux state store.

File: src/AutosizeInput.js [hunk 1]
```diff
@@ -32,6 +32,13 @@ const AutosizeInput = React.createClass({
 		this.copyInputStyles();
 		this.updateInputWidth();
 	},
+	shouldComponentUpdate (nextProps) {
+		// if this is the first time
+		if (nextProps === this.props) {
+			return true;
+		}
+		return nextProps.value !== this.props.value;
```
Comment (JedWatson) [inline src/AutosizeInput.js]: this would break component updates if any other props change, which is going to cause a problem :(

**Comment by stevemao** [comment]
@JedWatson @atran Can you guys have a look at #49 please.

**Pull request closed without merging**